syncer/syncer.go

```diff
@@ -411,7 +414,13 @@ func (s *Syncer) acceptLoop() error {
 	}
 }
 
-func (s *Syncer) peerLoop(closeChan <-chan struct{}) error {
+func (s *Syncer) isStopped() bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.l == nil
```

Comment:
Considering that we have a `shutdownCtx` as a signal now I'd probably use that instead of relying on the listener being `nil`.

e.g.
```go
select {
  case <-s.shutdownCtx:
    return true
  default:
    return false
}
```

Reply:
I considered that but didn't want to change it because of the slight nuance, this would return true before the listener gets closed. It's likely fine though I'll update it 👍